Makefile

```diff
@@ -27,9 +27,9 @@ openapi_file::
 	test -f $(OPENAPI_FILE) || curl -s -L $(SWAGGER_URL) > $(OPENAPI_FILE)
 
 ensure::
-	cd provider && go mod tidy
-	cd sdk && go mod tidy
-	cd tests && go mod tidy
+	cd provider && go mod tidy -compat=1.17
```

Comment:
Could we split up the go 1.17 bump to a separate PR? This change within the context of adding __pulumiType  seems good!